src/FMData.Rest.Auth.Cloud/FileMakerCloudAuthTokenProvider.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System.Net.Http.Headers;
+using System.Threading.Tasks;
+using Amazon.CognitoIdentityProvider;
+using Amazon.Extensions.CognitoAuthentication;
+using Amazon.Runtime;
+
+namespace FMData.Rest
+{
+    /// <summary>
+    /// FileMaker Cloud authentication provider (AWS Cognito)
+    /// </summary>
+    public class FileMakerCloudAuthTokenProvider : IAuthTokenProvider
+    {
+        private readonly ConnectionInfo _conn;
+
+        /// <summary>
+        /// Constructor 
+        /// </summary>
+        /// <param name="conn">Connection config values</param>
+        public FileMakerCloudAuthTokenProvider(ConnectionInfo conn)
+        {
+            _conn = conn;
+        }
+
+        /// <summary>
+        /// Connection config values
+        /// </summary>
+        public ConnectionInfo Conn { get => _conn; }
+
+        /// <summary>
+        /// Gets the AuthenticationHeaderValue
+        /// </summary>
+        /// <returns>AuthenticationHeaderValue</returns>
+        public async Task<AuthenticationHeaderValue> GetAuthenticationHeaderValue()
+        {
+            string token = await GetToken().ConfigureAwait(false);
+            return AuthenticationHeaderValue.Parse("FMID " + token);
+        }
+
+        /// <summary>
+        /// Provide an AWS Cognito Identiy Token
+        /// </summary>
+        private async Task<string> GetToken()
+        {
+            var provider = new AmazonCognitoIdentityProviderClient(new AnonymousAWSCredentials(), Amazon.RegionEndpoint.USWest2);
```

Comment:
I am not familair with Amazon Cognito, but I'm wondering if it would make sense for these two should be optional parameters for the FileMakerCloudAuthTokenProvider? 

That way, users can specify a different region and credential type.

Reply:
This refactor looks good. I think it makes sense this way, to lookup the parameters based on the `RegionEndpoint`. 

If someone needs something other than an `AnonymousAWSCredentials` type we can address that when it comes up.